vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java

```diff
@@ -493,6 +505,517 @@ protected void fireUpdateProgress(long totalBytes, long contentLength) {
         fireEvent(new ProgressUpdateEvent(this, totalBytes, contentLength));
     }
 
+    @DomEvent("file-reject")
+    public static class FileRejectEvent extends ComponentEvent<Upload> {
+        private final JsonObject detail;
+        private final JsonObject detailFile;
+        private final String detailError;
+
+        public FileRejectEvent(Upload source, boolean fromClient,
+                @EventData("event.detail") JsonObject detail,
+                @EventData("event.detail.file") JsonObject detailFile,
+                @EventData("event.detail.error") String detailError) {
+            super(source, fromClient);
+            this.detail = detail;
+            this.detailFile = detailFile;
+            this.detailError = detailError;
+        }
+
+        public JsonObject getDetail() {
+            return detail;
+        }
+
+        public JsonObject getDetailFile() {
+            return detailFile;
+        }
+
+        public String getDetailError() {
+            return detailError;
+        }
+    }
+
+    /**
+     * Adds a listener for {@code file-reject} events fired by the webcomponent.
+     *
+     * @param listener
+     *            the listener
+     * @return a {@link Registration} for removing the event listener
+     */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    protected Registration addFileRejectListener(
+            ComponentEventListener<FileRejectEvent> listener) {
+        return addListener(FileRejectEvent.class,
+                (ComponentEventListener) listener);
+    }
+
+    @DomEvent("upload-abort")
+    public static class UploadAbortEvent extends ComponentEvent<Upload> {
+        private final JsonObject detail;
+        private final JsonObject detailXhr;
+        private final JsonObject detailFile;
+
+        public UploadAbortEvent(Upload source, boolean fromClient,
+                @EventData("event.detail") JsonObject detail,
+                @EventData("event.detail.xhr") JsonObject detailXhr,
+                @EventData("event.detail.file") JsonObject detailFile) {
+            super(source, fromClient);
+            this.detail = detail;
+            this.detailXhr = detailXhr;
+            this.detailFile = detailFile;
+        }
+
+        public JsonObject getDetail() {
+            return detail;
+        }
+
+        public JsonObject getDetailXhr() {
+            return detailXhr;
+        }
+
+        public JsonObject getDetailFile() {
+            return detailFile;
+        }
+    }
+
+    /**
+     * Adds a listener for {@code upload-abort} events fired by the
+     * webcomponent.
+     *
+     * @param listener
+     *            the listener
+     * @return a {@link Registration} for removing the event listener
+     */
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    protected Registration addUploadAbortListener(
```

Comment:
Could be `private` since it's only used internally

Reply:
Removed as suggested by https://github.com/vaadin/flow-components/pull/4364#discussion_r1050476577